pg_search/src/index/reader/index.rs

```diff
@@ -912,6 +922,181 @@ impl SearchIndexReader {
+
+        if segment_count == 1 {
+            // Single segment: estimate directly without scaling
+            let segment_reader = self.searcher.segment_reader(0);
+            let segment_doc_proportion = segment_reader.num_docs() as f64 / total_docs;
+            self.estimate_node_recursive(
+                query_tree,
+                segment_reader,
+                segment_doc_proportion,
+                parser,
+            );
+        } else {
+            // Multiple segments (EXPLAIN ANALYZE): estimate on all segments and sum
+            self.estimate_multi_segment_recursive(query_tree, parser);
+        }
+    }
+
+    fn estimate_node_recursive<QueryParserCtor: Fn() -> QueryParser>(
+        &self,
+        node: &mut QueryWithEstimates,
+        largest_reader: &SegmentReader,
+        segment_doc_proportion: f64,
+        parser: &QueryParserCtor,
+    ) {
+        for child in node.children_mut() {
+            self.estimate_node_recursive(child, largest_reader, segment_doc_proportion, parser);
+        }
+
+        let tantivy_query_result = node.query.clone().into_tantivy_query(
+            &self.schema,
+            parser,
+            &self.searcher,
+            self.index_rel.oid(),
+            self.index_rel.rel_oid(),
+            None,
+            None,
+        );
+
+        match tantivy_query_result {
+            Ok(tantivy_query) => {
+                let weight = tantivy_query
+                    .weight(EnableScoring::Disabled {
+                        schema: self.schema.tantivy_schema(),
+                        searcher_opt: Some(&self.searcher),
+                    })
+                    .ok();
+
+                if let Some(weight) = weight {
+                    if let Ok(mut scorer) = weight.scorer(largest_reader, 1.0) {
+                        let mut count = scorer.size_hint() as usize;
+                        if count == 0 {
+                            count = scorer.count_including_deleted() as usize;
+                        }
+
+                        let estimated = if segment_doc_proportion > 0.0 {
+                            (count as f64 / segment_doc_proportion).ceil() as usize
+                        } else {
+                            count
+                        };
+
+                        node.set_estimate(estimated);
+                    } else {
+                        node.estimated_docs = None;
+                    }
+                } else {
+                    node.estimated_docs = None;
+                }
+            }
+            Err(_) => {
+                node.estimated_docs = None;
+            }
+        }
+    }
+
+    /// Estimate docs across multiple segments by summing counts from all segments.
+    /// This is used for EXPLAIN ANALYZE where all segments are open.
+    fn estimate_multi_segment_recursive<QueryParserCtor: Fn() -> QueryParser>(
+        &self,
+        node: &mut QueryWithEstimates,
+        parser: &QueryParserCtor,
+    ) {
```

Comment:
I don't think that the increase in accuracy is worth duplicating this method: instead, would it be possible to have both methods use only the largest segment?